external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1312,33 +1398,148 @@ fn path_escape(p: &Path) -> Result<String, fmt::Error> {
+    pkg_path: String,
+    dependencies: Vec<(&Dependency, PM::PackageName, &Package)>,
+) -> String {
+    let mut s = format!("\nAt {}", pkg_path).to_string();
     if !dependencies.is_empty() {
-        s.push_str(msg);
-        for (dep, pkg) in dependencies {
+        for (dep, pkg_name, pkg) in dependencies {
             s.push_str("\n\t");
-            s.push_str(format!("{}", DependencyTOML(pkg, dep)).as_str());
+            s.push_str(&format!("{pkg_name} = "));
+            s.push_str("{ ");
+            s.push_str(&format!("{pkg}"));
+            if let Some(digest) = dep.digest {
+                s.push_str(&format!(", digest = {digest}"));
+            }
+            if let Some(subst) = &dep.subst {
+                s.push_str(&format!(", addr_subst = {}", SubstTOML(subst)));
+            }
+            s.push_str(" }");
         }
+    } else {
+        s.push_str("\n\tno dependencies");
     }
     s
 }
 
-/// Check if dependencies of a given package in two different dependency graph maps are the same.
-fn pkg_deps_equal<'a>(
+/// Checks if dependencies of a given package in two different dependency graph maps are the same,
+/// checking both the dependency in the graph and the destination package (both can be
+/// different). Returns Ok(()) if they are and the two parts of the symmetric different between
+/// dependencies inside Err if they aren't.
+fn deps_equal<'a>(
     pkg_name: Symbol,
-    pkg_graph: &'a DiGraphMap<PM::PackageName, Dependency>,
-    other_graph: &'a DiGraphMap<PM::PackageName, Dependency>,
-) -> (
-    Vec<(&'a Dependency, PM::PackageName)>,
-    Vec<(&'a Dependency, PM::PackageName)>,
-) {
-    let pkg_edges = BTreeSet::from_iter(pkg_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
-    let other_edges =
-        BTreeSet::from_iter(other_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
-
-    let (pkg_deps, other_deps): (Vec<_>, Vec<_>) = pkg_edges
-        .symmetric_difference(&other_edges)
-        .partition(|dep| pkg_edges.contains(dep));
-    (pkg_deps, other_deps)
+    combined_graph: &'a DependencyGraph,
+    sub_graph: &'a DependencyGraph,
+) -> std::result::Result<
+    (),
+    (
+        Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
+        Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
+    ),
+> {
+    let combined_edges = BTreeSet::from_iter(
+        combined_graph
+            .package_graph
+            .edges(pkg_name)
+            .map(|(_, pkg, dep)| (dep, pkg, combined_graph.package_table.get(&pkg).unwrap())),
+    );
+    let sub_pkg_edges = BTreeSet::from_iter(
+        sub_graph
+            .package_graph
+            .edges(pkg_name)
+            .map(|(_, pkg, dep)| (dep, pkg, sub_graph.package_table.get(&pkg).unwrap())),
+    );
+
+    let (combined_pkgs, sub_pkgs): (Vec<_>, Vec<_>) = combined_edges
+        .symmetric_difference(&sub_pkg_edges)
+        .partition(|dep| combined_edges.contains(dep));
+    if combined_pkgs == sub_pkgs {
```

Comment:
It seems like you really like this pattern for detecting if both sides of the partition are empty ;) my personal preference is

```suggestion
    if combined_pkgs.is_empty() && sub_pkgs.is_empty() {
```

But I would also settle for a comment explaining that the two vector are only equal if they are both empty.

Reply:
Made the code change!